# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: uriloader/preload/PreloadHashKey.cpp
# Commit: 7d5683180ac3
# Full Hash: 7d5683180ac3f9e7cc96d68dde27b8b48642a7ac
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r a48aff5f2555 -r 7d5683180ac3 uriloader/preload/PreloadHashKey.cpp
--- a/uriloader/preload/PreloadHashKey.cpp	Thu Feb 17 15:57:52 2022 +0000
+++ b/uriloader/preload/PreloadHashKey.cpp	Thu Feb 17 15:57:52 2022 +0000
@@ -75,8 +75,8 @@
 }
 
 // static
-PreloadHashKey PreloadHashKey::CreateAsScript(nsIURI* aURI, CORSMode aCORSMode,
-                                              dom::ScriptKind aScriptKind) {
+PreloadHashKey PreloadHashKey::CreateAsScript(
+    nsIURI* aURI, CORSMode aCORSMode, JS::loader::ScriptKind aScriptKind) {
   PreloadHashKey key(aURI, ResourceType::SCRIPT);
   key.mCORSMode = aCORSMode;
 
@@ -89,9 +89,9 @@
 PreloadHashKey PreloadHashKey::CreateAsScript(nsIURI* aURI,
                                               const nsAString& aCrossOrigin,
                                               const nsAString& aType) {
-  dom::ScriptKind scriptKind = dom::ScriptKind::eClassic;
+  JS::loader::ScriptKind scriptKind = JS::loader::ScriptKind::eClassic;
   if (aType.LowerCaseEqualsASCII("module")) {
-    scriptKind = dom::ScriptKind::eModule;
+    scriptKind = JS::loader::ScriptKind::eModule;
   }
   CORSMode crossOrigin = dom::Element::StringToCORSMode(aCrossOrigin);
   return CreateAsScript(aURI, crossOrigin, scriptKind);